File: cpp/armstrong-numbers/armstrong_numbers.cpp
#include "armstrong_numbers.h"
#include <cmath>

namespace armstrong_numbers {

    bool is_armstrong_number(int n) {
        int num = n;
        int power = int(std::log10(n)) + 1;
        int sum = 0;
        while (n > 0) {
            sum += int(std::pow(n % 10, power));
            n /= 10;
        }
        return sum == num;
    }
}  // namespace armstrong_numbers

File: java/armstrong-numbers/src/main/java/ArmstrongNumbers.java
import java.util.Arrays;

class ArmstrongNumbers {

    boolean isArmstrongNumber(int numberToCheck) {
        String num = "" + numberToCheck;
        int n = num.length();
        int sum = 0;
        for (char c : num.toCharArray())
            sum += Math.pow(c - '0', n);
        return sum == numberToCheck;
    }

}
